src/mysql.c

```diff
@@ -472,6 +472,21 @@ static int mysql_read_slave_stats(mysql_database_t *db, MYSQL *con) {
     unsigned long long counter;
     double gauge;
 
+    if (row[SLAVE_SQL_RUNNING_IDX] == NULL) {
+      gauge_submit("slave_running", "sql", 1, db);
+    } else if (strcasecmp(row[SLAVE_SQL_RUNNING_IDX], "yes") != 0) {
+      gauge_submit("slave_running", "sql", 1, db);
+    } else {
+      gauge_submit("slave_running", "sql", 0, db);
+    }
```

Comment:
I think this could be condensed to (here and below):

```c
gauge_submit("bool", "slave-sql-running",
    (row[SLAVE_SQL_RUNNING_IDX] != NULL) && (strcasecmp(row[SLAVE_SQL_RUNNING_IDX], "yes") == 0),
    db);
```

This works, because a boolean in C is an integer that can only hold 0 and 1. It can implicitly we widened to a double, which is what this is taking advantage of. For the different metric name, see below.

Reply:
I have adjusted the code with your recommendation and built rpm packages I'm now going to test.